src/common/framework/tree.ts

```diff
@@ -282,7 +286,11 @@ function addSubtreeForTestPath(
         tree.query.filePathParts,
         subqueryTest
```

Comment:
I noticed here that we're passing a reference to `subqueryTest` into different `TestQueryMultiTest` objects, and appending to the array as we go. That doesn't seem like what we want?

Nevermind, I see the constructor makes a copy. Maybe we should make a copy before constructing otherwise callsites like this are a bit confusing.

This one could be:
```typescript
for (const [i, part] of test.entries()) {
  new TestQueryMultiTest(..., test.slice(0, i + 1));
}
```

And at [1] we would just pass `test` because at that point I think the arrays are the same.

Reply:
It is confusing, but it's also less error prone. I did it this way so I could be sure I wasn't missing any places that needed copies.

Doing the copy in the constructor makes the TestQuery classes immutable by construction, which reduces cognitive load working with them.